### PR TITLE
Add tests for BddScenariosCounter

### PR DESCRIPTION
**Pull request by ivan-lednev**

### CODE REVIEW

File: vividus/src/test/java/org/vividus/runner/BddScenariosCounterTests.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        root.addChild(beforeStories);
+        root.addChild(story);
+        story.addChild(beforeStory);
+        story.addChild(scenario);
+
+        when(runner.getDescription()).thenReturn(root);
+
+        BddScenariosCounter.main(new String[0]);
+
+        assertTrue(getOutput().contains(String.format(FORMATTER, 1, STORIES)));
```
Comment (valfirst) [inline vividus/src/test/java/org/vividus/runner/BddScenariosCounterTests.java]: use `assertThat` + corresponding matcher, it makes assertions more readable

Comment (valfirst) [reply vividus/src/test/java/org/vividus/runner/BddScenariosCounterTests.java]: here and below in similar cases

Comment (valfirst) [inline vividus/src/test/java/org/vividus/runner/BddScenariosCounterTests.java]: `String.format(FORMATTER, 1, STORIES)` -> replace with a plain string:
- there is no much sense to use `String.format` with constants only
- plain strings make unit tests more readable

File: vividus/src/test/java/org/vividus/runner/BddScenariosCounterTests.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    }
+
+    @Test
+    public void testHelpOptionIsPresent() throws Exception
+    {
+        BddScenariosCounter.main(new String[] {"--help"});
+        assertTrue(getOutput().contains("usage: BddScenariosCounter"));
+    }
+
+    private void stubPropertiesAndRunnerInvocations() throws Exception
```
Comment (valfirst) [inline vividus/src/test/java/org/vividus/runner/BddScenariosCounterTests.java]: https://martinfowler.com/articles/mocksArentStubs.html#TheDifferenceBetweenMocksAndStubs

Comment (valfirst) [inline vividus/src/test/java/org/vividus/runner/BddScenariosCounterTests.java]: `...And...` in the method name usually means that the method does more than one thing, but more than one thing is a violation of [Single responsibility principle](https://stackoverflow.com/questions/46541197/does-the-single-responsibility-principle-work-in-oop/46556381#46556381).  So what can be done here:
- `when(BeanFactory...` can't be used without `PowerMockito.mockStatic(BeanFactory.class);` so they can be grouped and placed into a new method
- mock `runner` is not used in all tests, so it makes sense to move its creation and `whenNew(JUnitReportingRunner...` to another method and this method should return just created mock:
```java
private JUnitReportingRunner mockJUnitReportingRunnerInstantiation()
{
    JUnitReportingRunner runner = mock(JUnitReportingRunner.class);
    whenNew(JUnitReportingRunner.class).withArguments(StoriesRunner.class).thenReturn(runner);
    return runner;
}
```
```

File: vividus/src/test/java/org/vividus/runner/BddStepsCounterTests.java [hunk 3]
```diff
@@ -52,6 +53,7 @@
 
 @RunWith(PowerMockRunner.class)
 @PrepareForTest({ Vividus.class, BeanFactory.class })
+@PowerMockIgnore({ "javax.*", "com.sun.*", "jdk.*", "org.xml.*", "org.w3c.*" })
```
Comment (valfirst) [inline vividus/src/test/java/org/vividus/runner/BddStepsCounterTests.java]: this change should go as a separate commit

File: vividus/src/test/java/org/vividus/runner/BddScenariosCounterTests.java [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    private static final String STORY = "story description";
+    private static final String SCENARIO = "scenario description";
+    private static final String EXAMPLE = "example description";
+    private static final String BEFORE = "before description";
+    private static final String RESOURCE_LOCATION = "bdd.story-loader.batch1.resource-location";
+
+    @Before
+    public void before()
+    {
+        PowerMockito.mockStatic(Vividus.class);
```
Comment (valfirst) [inline vividus/src/test/java/org/vividus/runner/BddScenariosCounterTests.java]: is it required?

Comment (ivan-lednev) [reply vividus/src/test/java/org/vividus/runner/BddScenariosCounterTests.java]: Without it, I get this exception
`java.lang.IllegalStateException: environments is not set`
